### ISSUE TITLE
Improve error message when Python version conflicts with package dependencies?

### ISSUE THREAD

**Issue by krinsman** [mamba-org/mamba#1002]
MWE: (assumes base conda is running Python 3.9 or Python 3.8, so created environment has that too)

```
conda create --name test
conda activate test
conda config --env --add channels bioconda
conda config --env --add channels conda-forge
conda install --yes mamba
mamba install pasta
```

The resulting error message (since the package as of this writing has evidently not been updated to use higher Python versions)

```
Encountered problems while solving:
  - nothing provides pymongo 3.3.0 py27_0 needed by pasta-1.7.8-0
```

This makes it sound like the problem is with conda-forge not providing the package pymongo. But when running

```
conda install pasta
```

one gets the real source of the error

```
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - pasta -> python[version='>=2.7,<2.8.0a0|>=3.7,<3.8.0a0|>=3.6,<3.7.0a0|>=3.5,<3.6.0a0|>=2.7,<3']

Your python: python=3.9

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.
```

Personally I think the error message from conda is more informative here than that from mamba. But if you disagree then this is a non-issue. I just thought it might help to be brought to your attention if you weren't already aware.

**Comment by adriendelsalle** [comment]
Hey @krinsman thank you for opening this issue!

I agree this is not informative enough.
The logs contain a `Pinning Python to '...'` message that could be helpful to also display at console.
I could be also pretty useful to be able to disable Python automatic pinning using configuration (`--no-py-pin`/`--py-pin`)

**Comment by adriendelsalle** [comment]
I just edited my answer because the solution proposed will maybe not work on `mamba` but only on `micromamba`.

In the PR opened, I propose at least to add python pin to the list of pinned packages in `mamba` (and do the exact same in `micromamba`):
```

Looking for: ['pasta']

bioconda/linux-64        Using cache
bioconda/noarch          Using cache
conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache

Pinned packages:
  - python 3.9.2

Encountered problems while solving:
  - nothing provides pymongo 3.3.0 py27_0 needed by pasta-1.7.8-0
```
We still need to improve `libsolv` problem itself

**Comment by h-vetinari** [comment]
> For better error messaging it might be worthwhile to get inspiration from pubgrub: https://medium.com/@nex3/pubgrub-2fb6470504f

Would it be realistic to change the backend to [pubgrub](https://github.com/pubgrub-rs/pubgrub)? Compared to the current conda/mamba resolution errors, this reads like a dream:
```
Because dropdown >=2.0.0 depends on icons >=2.0.0 and
  root depends on icons <2.0.0, dropdown >=2.0.0 is forbidden.

And because menu >=1.1.0 depends on dropdown >=2.0.0,
  menu >=1.1.0 is forbidden.

And because menu <1.1.0 depends on dropdown >=1.0.0 <2.0.0
  which depends on intl <4.0.0, every version of menu
  requires intl <4.0.0.

So, because root depends on both menu >=1.0.0 and intl >=5.0.0,
  version solving failed.
```

**Comment by wolfv** [comment]
yeah, it would at least be a really interesting experiment :) the pubgrub-rs docs make it sound like it could be quite straight-forward to try it on some conda repodata ...

I think we're quite invested into libsolv so maybe we'd first explore if we can add that functionality to the current code base of libsolv.

**Comment by h-vetinari** [comment]
@jonashaag, could you leave some breadcrumbs (commit, PR, docs, or even just a comment) when you close old issues?

**Comment by jonashaag** [comment]
Sorry! Closed this because of inactivity.

**Comment by h-vetinari** [comment]
Then it doubly needs a comment. I thought the issues had been fixed in the meantime...

I get that people want to have a "clean" issue tracker, but closing unsolved issues is very problematic IMO, even moreso without giving people a chance to comment if it's still relevant, much less closing without comment.

**Comment by jonashaag** [comment]
We have better error messages now but I don't know if this particular problem was fixed
